### ISSUE TITLE
Add support for Fb4 datatypes

### ISSUE THREAD

**Issue by arvanus** [mariuz/flamerobin#166]
New datatypes:

- INT128
- DECFLOAT(16)
- DECFLOAT(34)
- DECFLOAT
- TIME [ { WITHOUT | WITH } TIME ZONE ]
- TIMESTAMP [ { WITHOUT | WITH } TIME ZONE ]

**Comment by arvanus** [comment]
@andy-123 could you please take a look at Fb4 data types? I've found a bug and would like to ask you to give a look please

Steps:
```

CREATE TABLE TABLE_NAME (
    COLUMN_NAME  DECIMAL(18,6) NOT NULL,
    DEC_F2       NUMERIC(38,6)
);
ALTER TABLE TABLE_NAME ADD CONSTRAINT CONSTRAINT_NAME_PK PRIMARY KEY (COLUMN_NAME);
INSERT INTO TABLE_NAME (COLUMN_NAME, DEC_F2) VALUES (1, '123123124123412577766.441231');
select * from table_name;
--Edit manually dec_f2 to value 123123124123412577766.44123
--See that the value changed to 123123124123412577767.444123

```

Thanks

**Comment by andy-123** [comment]
See fix in PR #306
